### PR TITLE
Add a Makefile and cloudbuild.yaml

### PR DESCRIPTION
**Pull request by peterzhongyi**
Add a Makefile and cloudbuild.yaml to allow for building and pushing images of Cloud Armor Node Controller from source.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Build a docker image for the server, and tag it
+build:
+	docker build --tag=$(server_tag) .
+
+push: build
+	docker push $(server_tag)
+
+# build and push the example custom controller image with specified tag
+cloud-build:
+	gcloud builds submit --config=cloudbuild.yaml
```
Comment (zmerlynn) [inline Makefile]: I would add `--substitutions _REPOSITORY=$(REPOSITORY)`, then you can just do `make REPOSITORY=<blah> cloud-build`.

File: cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    id: push
+    dir: .
+    env: ['REPOSITORY=${_REPOSITORY}']
+    script: |
+      make push
+options:
+  dynamic_substitutions: true
+  defaultLogsBucketBehavior: REGIONAL_USER_OWNED_BUCKET
+substitutions:
+  _REPOSITORY: YOUR_REPOSITORY_HERE
```
Comment (zmerlynn) [inline cloudbuild.yaml]: I _think_ if you just remove the `_REPOSITORY` substition, the customer would have to provide the substitution on the command line, which would make this file generic.

File: Makefile (hunk 1 above)
Comment (peterzhongyi) [reply Makefile]: Done

File: cloudbuild.yaml (hunk 2 above)
Comment (peterzhongyi) [reply cloudbuild.yaml]: ohhhh that's better indeed

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**